File: allowance-questboad/lib/domain/repository/member/member_repository.dart
import '../../model/family/family_id.dart';
import '../../model/member/member.dart';
import '../../model/member/value_object/member_id.dart';

/// [Member]のリポジトリインターフェース
abstract interface class MemberRepository {
  /// 指定した[MemberId]に対応する[Member]を取得する
  Future<Member?> find(MemberId memberId);

  /// 指定した[FamilyId]に対応する[Member]を取得する
  Future<List<Member>?> findMembers(FamilyId familyId);
}

File: modules/allowance-questboard-edge/supabase/functions/member/repository/member_repository.ts
import { UserId } from "../../user/domain/value_object/user_id.ts";
import { Members } from "../domain/members.ts";
import { Member } from "../domain/member.ts";
import { MemberId } from "../domain/value_object/member_id.ts";
import { FamilyId } from "../../family/domain/value_object/family_id.ts";

export class MemberRepository {
    findAll(): Promise<Members> {
        throw new Error("Method findAll() must be implemented.");
    }
    findById(familyId: MemberId): Promise<Member | null> {
        throw new Error("Method findById() must be implemented.");
    }
    findByUserId(userId: UserId): Promise<Member | null> {
        throw new Error("Method findById() must be implemented.");
    }
    findByFamilyId(familyId: FamilyId): Promise<Members> {
        throw new Error("Method findById() must be implemented.");
    }
}
